### DISCUSSION TITLE
Force Spelling Errors

### DISCUSSION THREAD

**Discussion by LeStarch** [nasa/fprime#855] [Q&A] [answered]
@jsoref is there a way to force your spelling action to always flag a token as misspelled?  Specifically `m_data` is used often, but breaks builds against certain embedded OSes.  We don't have the licensing to build/ci these OSes but forcing the user to use another token would reduce the disruptions by about 90%.

Is this possible given your spelling action or am I looking at writing a specific action to target `m_data` usage?

**Comment by LeStarch** [comment, marked as answer]
Now that I think through this more, how would a contributor ever know this was the rationale and thus a needed fix?  This is an easy, but not effective solution.  Will consider a tailored action with specific messaging to let contributors know not to use m_data.

**Reply by jsoref** [reply]
Technically you could do a pattern something like `(?<=m_)d(?=ata)` that would cause the spell checker to lop off the `d` and then complain about `ata`.

But, indeed, I wouldn't recommend doing it -- because as you note, no one would be able to understand _why_ the spell checker is complaining about `ata`.

**Reply by jsoref** [reply]
I've added a feature note to myself for implementing something like this: https://github.com/check-spelling/check-spelling/wiki/Feature:-Reject-patterns

I don't expect to get to it in the near future. I have a number of other tasks I want to work on first.

**Reply by jsoref** [reply]
In case anyone runs into this discussion, this feature was released in [v0.0.20](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.20) as 🚫 [Forbidden patterns](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Forbidden-patterns), and apparently the specific vendor was [VxWorks](https://github.com/nasa/fprime/pull/797).

(I'm adding this comment now because i'm playing "connect the dots" -- people asked where the `m_data` ban came from...)

On the point of "how would the user know", that's actually a good point that my initial implementation didn't really handle, but a later feature (suggested patterns) does. Specifically, suggested patterns spits out the commented lines preceding a pattern when it suggests a pattern. I should see about having [forbidden patterns do the same](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Report-associated-comment-for-forbidden-patterns).
